Automattic/jetpack
# Sharing: do not include buttons in REST API responses

Fixes #7361

## Proposed changes:

As per the discussion in the related issue, there does not seem to be any obvious use-case where we would need to output sharing buttons in a REST API response, where we lack the other elements needed to display buttons properly (css and js).

This is a follow-up to #25346, where we already restricted where the sharing buttons were displayed.

### Other information:


## Jetpack product discussion

* N/A

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* Start by going to Jetpack > Settings > Social, and enable the sharing buttons.
* Go to Settings > Sharing, and change some of the settings. You can for example enable the buttons on more pages (archive pages, ...)
* View your site in different views, and ensure the buttons are displayed properly.
* Pick one of your posts, and query it via your site's REST API:
`https://yoursite.com/wp-json/wp/v2/posts/XXX` where `XXX` is the post ID.
* In the `content` field, you should not see the sharing buttons' markup.

You can also check for regressions in WoA sites:

* Configure sharing buttons to display them on "Front Page, Archive Pages, and Search Results", "Posts", and "Pages"
* Add a new page with a Blog Posts block (and not Posts Lists)
* Confirm sharing buttons are not displayed under each post in the list of posts in the block shown in the editor
